detox/test/e2e/16.location.test.js

```diff
@@ -1,45 +1,61 @@
-const exec = require('child-process-promise').exec;
-
-//TODO: Ignoring the test in CI until fbsimctl supports Xcode 9
-async function isFbsimctlInstalled() {
-  try {
-    await exec(`which fbsimctl`);
-    return true;
-  } catch (e) {
-    console.log(`setLocation only works through fbsimctl currently`);
-    return false;
-  }
-}
+const { expectToThrow } = require('./utils/custom-expects');
 
-describe('location', () => {
-  const lat = -80.125;
-  const lon = 66.5;
+// Skipped on Android because there is no Android permissions support yet
+describe(':ios: location request', () => {
+  it('should show nothing in the app (permission request)', async () => {
+    await device.relaunchApp({ permissions: { location: 'unset' } });
+
+    await getLocation();
+
+    await expect(element(by.id('error'))).not.toBeVisible();
+    await expectToThrow(() => waitFor(element(by.id(`latitude`))).toBeVisible().withTimeout(5500));
+  });
 
-  // Skipped on Android because there is no Android permissions support yet
-  it(':ios: Location should be unavailable', async () => {
-    if (!await isFbsimctlInstalled()) {
-      return;
-    }
+  it('should err when location permission was set to `never`', async () => {
     await device.relaunchApp({ permissions: { location: 'never' } });
-    await element(by.text('Location')).tap();
-    await element(by.id('getLocationButton')).tap();
-    await expect(element(by.id('error'))).toBeVisible();
+    await getLocation();
+
+    await expect(element(by.id('error'))).toHaveText('User denied access to location services.');
   });
+});
 
-  it('Should accept a location', async () => {
-    const isIOS = device.getPlatform() === 'ios';
+it('should allow to get and set location when location permission is set to `always`', async () => {
+  await device.relaunchApp({ permissions: { location: 'always' } });
 
-    if (isIOS && !await isFbsimctlInstalled()) {
-      return;
-    }
+  await expectLocationToBeAvailable();
+});
+
+it('should allow to get and set location when location permission is set to `inuse`', async () => {
+  await device.relaunchApp({ permissions: { location: 'inuse' } });
+  await expectLocationToBeAvailable();
+});
 
-    await device.relaunchApp({ permissions: { location: 'always' } });
-    await device.setLocation(lat, lon);
-    await element(by.text('Location')).tap();
-    await element(by.id('getLocationButton')).tap();
-    await waitFor(element(by.text(`Latitude: ${lat}`))).toBeVisible().withTimeout(5500);
+describe(':android: location permission granted from manifest', () => {
+  beforeEach(async () => {
+    await device.relaunchApp();
+  });
 
-    await expect(element(by.text(`Latitude: ${lat}`))).toBeVisible();
-    await expect(element(by.text(`Longitude: ${lon}`))).toBeVisible();
+  it('should allow to get and set location', async () => {
+    await expectLocationToBeAvailable();
   });
 });
+
+// Helpers
```

Comment:
The typical pattern is to use a _driver_. Example: https://github.com/wix/Detox/blob/master/detox/test/e2e/drivers/actions-driver.js

Reply:
done :+1: